pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -314,6 +323,15 @@ func ingestWithRetries(
 	return nil
 }
 
+func loadRetainedTime(ctx context.Context, db isql.DB, ingestionJob *jobs.Job) hlc.Timestamp {
+	latestProgress, err := replicationutils.LoadIngestionProgress(ctx, db, ingestionJob.ID())
```

Comment:
One note about using LoadIngestionProgress here rather than using a method based off of the `job` object is that we don't end up asserting anything about the lease.  But, I think that is OK for now.

Reply:
yeah, the `job` object really needs a "load this job info transactionally" public method